Homework tasks

bakhtiyar-bauyrzhan opened the pull request.

bakhtiyar-bauyrzhan committed:
Default environment

Added plane and capsule

bakhtiyar-bauyrzhan committed:
Added some mechanics

Added:
movement mechanics
rotation mechanics

Created:
Capsule with face that moves according to movement direction

Delt06 commented on Assets/Scripts/Movement.cs:
```diff
@@ -0,0 +1,27 @@
+        var movement = new Vector3(Input.GetAxis("Horizontal"), 0, Input.GetAxis("Vertical"));
+        transform.position += movement * Speed * Time.deltaTime;
+        
+        if (movement == Vector3.zero)
```
floating-point equality checks are not reliable. You should check whether the vector's magnitude is above the threshold

Delt06 commented on Assets/Scripts/Movement.cs:
```diff
@@ -0,0 +1,27 @@
+    void Update()
+    {
+        // movement direction
+        var movement = new Vector3(Input.GetAxis("Horizontal"), 0, Input.GetAxis("Vertical"));
```
the diagonal movement is faster than when moving along one of the axes
if movement vector's length is above 1, you should normalize it

Delt06 replied:
-0.5 points

Delt06 reviewed (approved):
## General Feedback
Nice controls, especially given how little code there is.

## Grade
4.5 + 1 (extra) + 0.5 (code quality) = 6 points

bakhtiyar-bauyrzhan committed:
Minor fixes

Fixed:
diagonal movement speed doesn't exceed using normalization
floating point equality changed to magnitude comparison
